File: web/src/app/service/type.service.ts
import { Injectable } from '@angular/core';
import {CommonService} from './common.service';
import {HttpClient} from '@angular/common/http';
import {Router} from '@angular/router';
import {Observable} from 'rxjs';
import {Page} from '../base/page';
import {Type} from '../func/Type';

@Injectable({
  providedIn: 'root'
})
export class TypeService {
  private url = 'type';
  constructor(private commonService: CommonService,
              private httpClient: HttpClient,
              private router: Router) { }

  /**
   * 分页方法
   * @param page 第几页
   * @param size 每页条数
   * @param userId 用户
   */
  public getAll(page: number, size: number): Observable<Page<Type>> {
    const params: { [key: string]: any } = {
      page: String(page),
      size: String(size),
    };

    return this.httpClient.get<Page<Type>>(`${this.url}/getAll`, {params});
  }

  save(type: Type): any {
    console.log(type);
    return this.httpClient.post(`${this.url}`, type);
  }

  public delete(typeId: number): Observable<null> {
    return this.httpClient.delete<null>(`${this.url}/${typeId.toString()}`);
  }

  findAll(): any{
    return this.httpClient.get<Array<Type>>(`${this.url}`);
  }

  /**
   * 通过Id获取用户
   */
  public getTypeById(typeId: number): Observable<Type> {
    return this.httpClient.get<Type>(`${this.url}/${typeId.toString()}`);
  }

  /**
   * 更新
   */
  public update(typeId: number, type: Type): Observable<Type> {
    return this.httpClient.put<Type>(`${this.url}/${typeId.toString()}`, type);
  }

}

File: api/src/main/java/equipmentManagementSystem/service/TypeService.java
package equipmentManagementSystem.service;

import equipmentManagementSystem.entity.Department;
import equipmentManagementSystem.entity.Equipment;
import equipmentManagementSystem.entity.Type;
import equipmentManagementSystem.entity.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface TypeService {
    Page<Type> findAll(Pageable pageable);

    void add(Type type);

    List<Type> getAll();
    /**
     * 删除
     *
     * @param id
     */
    void delete(Long id);

    Type update(Long id, Type type);

    Type getTypeById(Long id);
}
